sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -340,8 +340,11 @@ object GpuWindowExec extends Arm {
           case wf: GpuWindowFunction =>
             // All window functions, including those that are also aggregation functions, are
             // wrapped in a GpuWindowExpression, so dedup and save their children into the pre
-            // stage, replacing them with aliases.
-            val newChildren = wf.children.map(extractAndSave(_, preProject, preDedupe))
+            // stage, replacing them with aliases. Also, sometimes children are not provided in the
+            // initial list of expressions after optimizations, so we add them here, and they will
+            // be deduped anyways in the other passes
+            val newChildren = wf.children.map(ce =>
```

Comment:
@mythrocks  mind taking a look at this

Reply:
Sorry for the delay.

I'm not particularly familiar with the logic behind `isPreNeeded`, etc. But on discussion with @NVnavkumar, one wonders if we should check why `isPreNeeded` is turning up `false` on Databricks, with AQE turned on. We might adjust how `isPreNeeded` is calculated.

Reply:
So, `isPreNeeded` is `false` on Databricks, but it's a bit of a red herring in this case. We actually need to the window function children in the GpuWindowExec itself, it looks like the extra GpuProject does not help in this case.

Reply:
After some exploration, I have determined that the bug this fixes is not AQE-specific and I'm not 100% confident that this fix is currently the right approach. I have reverted this fix and filed a new issue https://github.com/NVIDIA/spark-rapids/issues/6531 to track the bug here.